tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -514,15 +514,20 @@ private[optimizer] abstract class OptimizerCore(
       case RecordValue(tpe, elems) =>
         RecordValue(tpe, elems map transformExpr)
 
-      case IsInstanceOf(expr, ClassType(ObjectClass)) =>
+      case IsInstanceOf(expr, ClassRef(ObjectClass)) =>
         transformExpr(BinaryOp(BinaryOp.!==, expr, Null()))
 
-      case IsInstanceOf(expr, tpe) =>
+      case IsInstanceOf(expr, typeRef) =>
         trampoline {
           pretransformExpr(expr) { texpr =>
             val result = {
-              // TODO This cast is suspicious
-              if (isSubtype(texpr.tpe.base, tpe.asInstanceOf[Type])) {
+              // TODO The hard conversion of ClassRef into ClassType is suspicious
```

Comment:
But problem is that not all `ClassRef`s have a corresponding `ClassType`. Things that don't:
* `ClassRef`s of primitive types
* `ClassRef`s of JS types

However now I think this I understand why this is valid here: because for all `ClassRef`s *that are admissible in an `Is/AsInstanceOf`*, their erased `Type` is indeed a `ClassType` (except `j.l.Object`, special-cased above).